include/kernel_helper.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "comm_quda.h"
+
+namespace quda {
+
+  struct kernel_t {
+    const void *func;
+    const std::string name;
+
+    kernel_t(const void *func, const char *name) :
+      func(func),
+      name(name) {}
+  };
+
+  template <bool use_kernel_arg_ = true>
+  struct kernel_param {
+    static constexpr bool use_kernel_arg = use_kernel_arg_;
+    dim3 threads; /** number of active threads required */
```

Comment:
is threads a good name here? I think in some place work_items was used. Maybe this is something like work_size ? Not sure yet.

Reply:
There was something called `work_items` that's used in the dslash packing.

I agree that `threads` isn't the best name, and something `work_items`, `work_size` could be more appropriate.  Changing this involves touching some 200 lines, so I ask that we all agree on a name, before I make this change.  I guess I like `work_items` best.

Reply:
@weinbe2 @hummingtree

Reply:
Discuss offline: we'll keep the name as `threads` for now, but may examine in the future.